dataland-frontend/tests/component/components/resources/frameworkDataSearch/p2p/P2pPanel.cy.ts

```diff
@@ -0,0 +1,202 @@
+    baseDataset: PathwaysToParisData
+  ): DataAndMetaInformationPathwaysToParisData[] {
+    const p2pDatasets: DataAndMetaInformationPathwaysToParisData[] = [];
+    for (let i = 0; i < 6; i++) {
+      const reportingYear = 2023 + i;
+      const reportingDate = `${reportingYear}-01-01`;
+      const p2pData = structuredClone(baseDataset) as PathwaysToParisData;
+      p2pData.general.general.dataDate = reportingDate;
+      const metaData: DataMetaInformation = {
+        dataId: `dataset-${i}`,
+        reportingPeriod: reportingYear.toString(),
+        qaStatus: QaStatus.Accepted,
+        currentlyActive: true,
+        dataType: DataTypeEnum.P2p,
+        companyId: "mock-company-id",
+        uploadTime: 0,
+        uploaderUserId: "mock-uploader-id",
+      };
+
+      p2pDatasets.push({
+        metaInfo: metaData,
+        data: p2pData,
+      });
+    }
+    return p2pDatasets;
+  }
+
+  it("Check P2p view page for company with six P2p data sets reported in different years ", () => {
+    const preparedFixture = getPreparedFixture("six-p2p-data-sets-in-different-years", preparedFixtures);
+    const mockedData = constructCompanyApiResponseForP2pForSixYears(preparedFixture.t);
+    cy.intercept("/api/data/p2p/companies/mock-company-id", mockedData);
+
+    cy.mountWithPlugins(P2pPanel, {
+      keycloak: minimalKeycloakMock({}),
+      data() {
+        return {
+          companyId: "mock-company-id",
+        };
+      },
+    });
+    cy.get(`span[data-test=General]`).click();
+    cy.get(`span[data-test=_general]`).click();
+    cy.get("table").find(`tr:contains("Data Date")`).find(`span`).eq(6).contains("2023");
+
+    for (let indexOfColumn = 1; indexOfColumn <= 6; indexOfColumn++) {
+      cy.get(`span.p-column-title`)
+        .eq(indexOfColumn)
+        .should("contain.text", (2029 - indexOfColumn).toString());
+    }
+  });
+
+  it("Unit test for sortReportingPeriodsToDisplayAsColumns", () => {
+    const firstYearObject = { dataId: "5", reportingPeriod: "2022" };
+    const secondYearObject = { dataId: "2", reportingPeriod: "2020" };
+    const firstOtherObject = { dataId: "3", reportingPeriod: "Q2-2020" };
+    const secondOtherObject = { dataId: "6", reportingPeriod: "Q3-2020" };
+    const boolList = [false, true]; //Apparently Typescript doesn't like type conversions, so input is direct.
+    for (let i = 0; i < 2; i++) {
+      expect(sortReportingPeriodsToDisplayAsColumns([secondYearObject, firstYearObject], boolList[i])).to.deep.equal([
+        firstYearObject,
+        secondYearObject,
+      ]);
+
+      expect(sortReportingPeriodsToDisplayAsColumns([secondOtherObject, firstOtherObject], boolList[i])).to.deep.equal([
+        firstOtherObject,
+        secondOtherObject,
+      ]);
+    }
+    expect(sortReportingPeriodsToDisplayAsColumns([firstYearObject, secondYearObject, firstYearObject])).to.deep.equal([
+      firstYearObject,
+      firstYearObject,
+      secondYearObject,
+    ]);
+    expect(
```

Comment:
What's the difference to the test above?

Reply:
Oh I just noticed that it was not firstYearObject twice. So we should actually keep it since it is untested behaviour else.

Reply:
@marcelrasbachdfine

Reply:
> Oh I just noticed that it was not firstYearObject twice. So we should actually keep it since it is untested behaviour else.

there was one function which actually had firstYearObject twice, but the other one doesn't.
I just outsourced it to another utils file... so i can just remove it in one of these?

Reply:
No need to create the function `sortReportingPeriodsToDisplayAsColumnsTest` because we should delete one of these unit tests calling this function entirely. They are exactly the same test.

Reply:
```
expect(sortReportingPeriodsToDisplayAsColumns([firstYearObject, secondYearObject, firstYearObject])).to.deep.equal([
      firstYearObject,
      firstYearObject,
      secondYearObject,
    ]);
```
This test went missing.

Reply:
Second from bottom.

Reply:
Isn't this test somehow a dublication of one of the others? Maybe let's also discuss it in the call

Reply:
Ok